### ISSUE TITLE
offer to renew link

### ISSUE THREAD

**Issue by Mike-Reddick** [jingyu9575/multithreaded-download-manager#118]
If have an failed download or an download which invalid link, like some youtube streaming link, is good an feature so, when you add an file with the same name, offer option to just renew the old entry, so, download can be resumed, an no need copy and paste links.

Thanks!

**Comment by jingyu9575** [comment]
You mean in the automatic prompt dialog? I think it is possible to implement.

**Comment by Mike-Reddick** [comment]
Yes in automatic prompt dialog, maybe an checkbox something like "renew link, instead of add a new one". But I'm thinking about and if implemented is good some conditions:

1- Only show this checkbox if have just one download entry with an file with the same name, if have two or more, don't show.

2- Only show chekbox if this download is in Failed state. Don't show check box if is Paused , Stopped (canceled), Running or Complete. 

3-  Only show checkbox if download is partially downloaded, don't show if it is 0% downloaded.

4- Only show checkbox if download can be resumed

Is a lot of conditions and work, is just suggestions... But you know what can be better and if implement it or not! Anyway thanks!
